runtime/compiler/control/J9CompilationStrategy.cpp

```diff
@@ -39,94 +39,22 @@
-bool                     TR::CompilationController::_tlsCompObjCreated = false;
-
-
-//------------------------------------ init -----------------------------------
-// Initializes the compilationController.
-// Return false if it fails
-//-----------------------------------------------------------------------------
-bool TR::CompilationController::init(TR::CompilationInfo *compInfo)
+TR::CompilationStrategy*
+J9::CompilationStrategy::createInstance(const char* name)
    {
-   _useController = false; // Default to failure
-   _compilationStrategy = 0; // Default to failure
-   TR::Options *options = TR::Options::getCmdLineOptions();
-   char *strategyName = options->getCompilationStrategyName();
-
-   if (strategyName && strcmp(strategyName, "none"))
-      {
-      _compInfo = compInfo;
-      if (strcmp(strategyName, "default") == 0)
-         _compilationStrategy = new (PERSISTENT_NEW) TR::DefaultCompilationStrategy();
-      // checks for other strategies go here
-      else // if no match, use default
-      {
-         _compilationStrategy = new (PERSISTENT_NEW) TR::DefaultCompilationStrategy();
-      }
-
-      if (_compilationStrategy)
-         {
-         TR_OptimizationPlan::_optimizationPlanMonitor = TR::Monitor::create("OptimizationPlanMonitor");
-         _useController = (TR_OptimizationPlan::_optimizationPlanMonitor != 0);
-         if (_useController)
-            {
-            static char *verboseController = feGetEnv("TR_VerboseController");
-            if (verboseController)
-               setVerbose(atoi(verboseController));
-            if (verbose() >= LEVEL1)
-               fprintf(stderr, "Using %s comp strategy\n", strategyName);
-            }
-         }
-      }
-   //TR_ASSERT(_useController, "Must use compilation controller");
-//#ifdef COMP_YIELD_ANALYSIS
-   if (options->getOption(TR_EnableCompYieldStats))
-      TR::Compilation::allocateCompYieldStatsMatrix();
-   tlsAlloc(OMR::compilation);
-   _tlsCompObjCreated = true;
-   return _useController;
-   }
-
-
-//-------------------------------- shutdown ---------------------------------
-// Called at shutdown time after compilation thread has been stopped
-// --------------------------------------------------------------------------
-void TR::CompilationController::shutdown()
-   {
-   if (_tlsCompObjCreated)
-      tlsFree(OMR::compilation);
-   if (!_useController)
-      return;
-   // would like to free all entries in the pool of compilation plans
-   int32_t remainingPlans = TR_OptimizationPlan::freeEntirePool();
-      // print some stats
-   if (verbose() >= LEVEL1)
-      {
-      fprintf(stderr, "Remaining optimizations plans in the system: %d\n", remainingPlans);
-      }
-   _compilationStrategy->shutdown();
+   J9::CompilationStrategy* instance = NULL;
+   instance = new (PERSISTENT_NEW) J9::CompilationStrategy();
+   return static_cast<TR::CompilationStrategy*>(instance);
```

Comment:
Would it work without the cast?

Reply:
Updated - remove static method createInstance now, add self method in OMRCompilationStrategy and use the method in J9CompilationStrategy